File: app_manager/appmanager/src/main/java/app/editors/manager/ui/adapters/diffutilscallback/FileDiffUtilsCallback.java
package app.editors.manager.ui.adapters.diffutilscallback;

import java.io.File;
import java.util.List;

public class FileDiffUtilsCallback extends BaseDiffUtilsCallback<File> {

    public FileDiffUtilsCallback(List<File> mNewList, List<File> mOldList) {
        super(mNewList, mOldList);
    }

    @Override
    public boolean areItemsTheSame(int oldItemPosition, int newItemPosition) {
        File oldFile = mOldList.get(oldItemPosition);
        File newFile = mNewList.get(newItemPosition);
        return oldFile.getAbsolutePath().equals(newFile.getAbsolutePath());
    }

    @Override
    public boolean areContentsTheSame(int oldItemPosition, int newItemPosition) {
        File oldFile = mOldList.get(oldItemPosition);
        File newFile = mNewList.get(newItemPosition);
        return oldFile.lastModified() == newFile.lastModified();
    }
}

File: app_manager/appmanager/src/main/java/app/editors/manager/ui/adapters/diffutilscallback/FileDiffUtilsCallback.kt
package app.editors.manager.ui.adapters.diffutilscallback

import java.io.File

class FileDiffUtilsCallback(
    newList: List<File>,
    oldList: List<File>
) : BaseDiffUtilsCallback<File>(newList, oldList) {

    override fun areItemsTheSame(oldItemPosition: Int, newItemPosition: Int): Boolean {
        val oldFile = mOldList[oldItemPosition]
        val newFile = mNewList[newItemPosition]
        return oldFile.absolutePath == newFile.absolutePath
    }

    override fun areContentsTheSame(oldItemPosition: Int, newItemPosition: Int): Boolean {
        val oldFile = mOldList[oldItemPosition]
        val newFile = mNewList[newItemPosition]
        return oldFile.lastModified() == newFile.lastModified()
    }
}